### ISSUE TITLE
🐛 [BUG]: edge.sourceNode is not updated by updateEdge when edge.source is updated

### ISSUE THREAD

**Issue by ljw20180420** [bcakmakoglu/vue-flow#1704]
### Is there an existing issue for this?

- [X] I have searched the existing issues and this is a new bug.

### Current Behavior

I use Events API.
```python
const { onEdgeUpdate, updateEdge } = useVueFlow();
onEdgeUpdate((params) => {
  updateEdge(params.edge, params.connection)
})
```
However, `updateEdge` only updates `edge.source` but not `edge.sourceNode`.

### Expected Behavior

Both `edge.source` but not `edge.sourceNode` should be updated.

### Steps To Reproduce

```bash
git clone git@github.com:ljw20180420/vueFlowBug.git
cd vueFlowBug
npm install
npm run dev
```
In the browser,
```list
Neglect the breakpoint at start up.
Move the source of the edge from `node 1` to `node3`.
In the console, `edge.source` becomes `node 3`, while `edge.sourceNode.id` is still `node 1`.
```

**Comment by bcakmakoglu** [comment]
Thanks for the report, will be fixed in the next patch.

**Comment by bcakmakoglu** [comment]
Fixed with `1.41.6`
